nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/jcamp/JCAMPData.java

```diff
@@ -840,4 +844,88 @@ public static boolean findData(StringBuilder bpath) {
         // FID and Dataset have the same extensions
         return findFID(bpath);
     }
+
+    /**
+     * Get the total size of a dimension including both real and imaginary.
+     * @param dim The dimension to use.
+     * @return The total size.
+     */
+    private int getTotalSize(int dim) {
+        int factor = isComplex(dim) ? 2: 1;
+        return getSize(dim) * factor;
+    }
+
+    /**
+     * Get a name to use for a dataset based on the provided filename.
+     * @param file The File object to parse the dataset name from.
+     * @return A String dataset name.
+     */
+    public String suggestName(File file) {
+        String fileName = file.getName();
+        int lastDot = fileName.lastIndexOf(".");
+        if (lastDot != -1) {
+            fileName = fileName.substring(0, lastDot);
+        }
+        return fileName;
+    }
+
+    /**
+     * Create a Dataset from the JCAMP data. This method assumes the JCAMP data has already
+     * been processed.
+     * @param datasetName The String name to use for the new Dataset.
+     * @return The newly created Dataset.
+     * @throws IOException
+     * @throws DatasetException
+     */
+    public Dataset toDataset(String datasetName) throws IOException, DatasetException {
+        File file = new File(path);
+        Path fpath = file.toPath();
+
+        int[] dimSizes = new int[getNDim()];
+        for(int i = 0; i < getNDim(); i++) {
+            dimSizes[i] = getTotalSize(i);
+        }
+
+        if (datasetName == null) {
+            datasetName = suggestName(fpath.toFile());
+        }
+        // Create a dataset in memory
+        Dataset dataset = new Dataset(datasetName + ".nv", dimSizes, true);
+        dataset.newHeader();
+        // Set the processed data into the dataset
+        boolean hasImaginaryData = this.imaginary.length != 0;
+        Vec complex;
+        for (int index = 0; index < this.real.length; index++){
+            if (hasImaginaryData) {
+                complex = new Vec(this.real[index], this.imaginary[index]);
+            } else {
+                complex = new Vec(this.real[index]);
+            }
+            complex.setFreqDomain(true);
+            dataset.writeVector(complex, index, 0);
+        }
+        // Set the header information in the dataset
+        for (int i = 0; i < dataset.getNDim(); i++) {
+            dataset.setValues(i, getValues(i));
+            // The first dimension is set when the vectors are set.
+            if (i > 0) {
+                dataset.setComplex(i, false);
+            }
```

Comment:
This method is based off of the RS2D.toDatasets method, there the complex mode was set using i==0, so the first dimensions was always complex and the other dimensions were always false. Had some difficulty trying to reach a solution where both the T1 and JRES datasets loaded properly (tried using i==0 and using isComplex(i)) so ended up using whatever writeVector on line 905 set complex for the first dimension and setting all other dimensions to always false. Not sure if this is acceptable?